python/lsst/meas/algorithms/testUtils.py

```diff
@@ -178,3 +184,121 @@ def makeDefectList():
                   ]
 
     return defectList
+
+
+class MockLoadReferenceObjects(ReferenceObjectLoader):
+    """A simple mock of LoadReferenceObjects for tests.
```

Comment:
I'm not sure this is a "simple" mock: it seems like there's a lot going on in here. You've effectively re-implemented most of the important parts of `ReferenceObjectLoader` in here and I'm concerned that this hides some of the things we want to test. I wonder if we could get by with some `unittest.mock`s of dataId/datasetRef instead? Or even have a very minimal butler live in `meas_algorithms/tests/data`?